mlflow/utils/_capture_transformers_modules.py

```diff
@@ -0,0 +1,80 @@
+"""
+This script should be executed in a fresh python interpreter process using `subprocess`.
+"""
+import json
+import os
+import sys
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.utils._capture_modules import _CaptureImportedModules, parse_args, store_imported_module
+
+
+class _CaptureImportedModulesForHF(_CaptureImportedModules):
+    """
+    A context manager to capture imported modules by temporarily applying a patch to
+    `builtins.__import__` and `importlib.import_module`.
+    Used for 'transformers' flavor only.
+    """
+
+    def __init__(self, module_to_throw):
+        super().__init__()
+        self.module_to_throw = module_to_throw
+
+    def _wrap_package(self, name):
+        if name == self.module_to_throw or name.startswith(f"{self.module_to_throw}."):
+            raise ImportError(f"Disabled package {name}")
+
+    def _record_imported_module(self, full_module_name):
+        self._wrap_package(full_module_name)
+        return super()._record_imported_module(full_module_name)
```

Comment:
```suggestion
    def _record_imported_module(self, full_module_name):
        if name == self.module_to_throw or name.startswith(f"{self.module_to_throw}."):
            raise ImportError(f"Disabled package {name}")
        return super()._record_imported_module(full_module_name)
```


`_wrap_package` sounds a bit confusing. I'd remove it.